lib/Bagger/Storage/Config.pm

```diff
@@ -0,0 +1,118 @@
+=head1 NAME
+
+   Bagger::Storage::Config -- Configuration Routines for Bagger Storage
+
+=cut
+
+package Bagger::Storage::Config;
+
+=head1 SYNOPSIS
+
+   my $config = Bagger::Storage::Config(
+             key   => 'additional_index_types', 
+	     value => ['rum', 'sp-gist']
+   );
+   $config->save
+
+=cut
+
+use strict;
+use warnings;
+use Carp 'croak';
+use Moose;
+use PGObject::Util::DBMethod;
+use PGObject::Type::JSON;
+use namespace::autoclean;
+use Scalar::Util 'reftype';
+use Moose::Util::TypeConstraints;
+with 'Bagger::Storage::PGObject';
+PGObject::Type::JSON->register; # become the codec for Perl <-> json
+                                # but not jsonb by default
+
+=head1 DESCRIPTION
+
+ The Bagger Storage Config module exists to configure anything needed regarding
+ Bagger storage nodes as a cluster.  It is expected that this data will somehow
+ be synchronized to the storage nodes themselves along with dimensions and
+ index information.
+
+ The configuration system is intended to be flexible and portable.  Values can
+ be anything, but undef is not accepted and a value must be provided.  Strings,
+ array and hashrefs, and references are all supported.
+
+ The data values are serialized to JSON in the database, which has one important
+ side effect.  If you pass a scalar reference to value, then when you retrieve
+ that reference, it will be returned as a scalar.
+
+=head1 PROPERTIES
+
+=over
+
+=item id Int, assigned by the database
+
+=cut
+
+has id => (is => 'ro', isa => 'Int');
+
+=item key Str, required
+
+=cut
+
+has key => (is => 'ro', isa => 'Str', required => 1);
+
+=item value -- Arbitrary scalar, hashref, or arrayref.  Required
+
+=item value_string -- For scalars, handle as string rather json conversion ref
+ 
+  use this one if you expect a string or number back.
+
+=cut
+
+subtype 'PGObject::Type::JSON'
+  => as 'PGObject::Type::JSON';
+
+coerce 'PGObject::Type::JSON'
+  => from 'Str | Ref'
+  =>  via { croak if (defined reftype($_)) and (reftype($_) =~ /CODE|SCALAR/);
+           return PGObject::Type::JSON->new($_) };
+
+
+has value => (is      => 'ro', 
+             isa      => 'PGObject::Type::JSON',
+             required => 1,
+	     coerce   => 1,
+             );
+sub value_string {
+    my $self = shift;
+    croak 'Expected a SCALAR config variable' 
+                              unless $self->value->reftype eq 'SCALAR';
+    return ${$self->value};
+}
+
+
+=head1 METHODS
+
+=head2 get($key)
+
+ Looks up the config item by key and returns it.
+
+=cut
+
+sub get {
+    my $self = shift;
+    my $key = (shift // $self);
```

Comment:
-Do we ever expect to call sub get{} without it being a method of an object?- 

Belay that, got it. We can instantiate an object via get.

Reply:
So this allows us to call it either as:
Bagger::Storage::Config::get('something');

or
Bagger::Storage::Config->get('something');

The difference is that in the second, the class name ends up in the first arg.